src/AI/Muriel.java

```diff
@@ -0,0 +1,750 @@
+    {
+      for( Commander oCo : enemyCos )
+      {
+        for( UnitModel otherModel : otherUnitModels.get(oCo) )
+        {
+          Unit myUnit = new Unit(myCo, myModel);
+          Unit otherUnit = new Unit( oCo, otherModel );
+
+          double myDamage = 0;
+          Weapon myWeapon = myUnit.chooseWeapon(otherUnit.model, 1, false);
+          if( null != myWeapon )
+          {
+            BattleInstance.BattleParams params = new BattleInstance.BattleParams(myUnit, myWeapon,
+                otherUnit, Environment.getTile(TerrainType.ROAD, Weathers.CLEAR));
+            myDamage = params.calculateDamage();
+          }
+
+          // Now go the other way.
+          double otherDamage = 0;
+          Weapon otherWeapon = otherUnit.chooseWeapon(myUnit.model, 1, false);
+          if( null != otherWeapon )
+          {
+            BattleInstance.BattleParams params = new BattleInstance.BattleParams(otherUnit, otherWeapon,
+                myUnit, Environment.getTile(TerrainType.ROAD, Weathers.CLEAR));
+            otherDamage = params.calculateDamage();
+          }
+
+          // Calculate and store the damage and cost-effectiveness ratios.
+          double ratio = 0;
+          double invRatio = 0;
+          if( myDamage != 0 && otherDamage != 0)
+          {
+            ratio = myDamage / otherDamage;
+            invRatio = 1/ratio;
+          }
+          if( myDamage == 0 ) ratio = 0;
+          if( otherDamage == 0 ) invRatio = 0;
+          if( myDamage != 0 && otherDamage == 0 ) ratio = Double.MAX_VALUE;
+          if( myDamage == 0 && otherDamage != 0 ) invRatio = Double.MAX_VALUE;
+          double myCostRatio = ratio * ((double)otherModel.moneyCost / myModel.moneyCost);
+          double otherCostRatio = invRatio * ((double)myModel.moneyCost / otherModel.moneyCost);
+          myUnitEffectMap.put(new UnitModelPair(myModel, otherModel), new UnitMatchupAndMetaInfo(ratio, myCostRatio));
+          myUnitEffectMap.put(new UnitModelPair(otherModel, myModel), new UnitMatchupAndMetaInfo(invRatio, otherCostRatio));
+
+          System.out.println(String.format("%s vs %s: %s/%s costRatio: %s", myUnit, otherUnit, myDamage, otherDamage, myCostRatio));
+        }
+      }
+    }
+  }
+
+  @Override
+  public void initTurn(GameMap gameMap)
+  {
+    turnNum++;
+    log(String.format("[======== Muriel initializing turn %s for %s =========]", turnNum, myCo));
+
+    // Make a list of properties we want to claim.
+    nonAlliedProperties = AIUtils.findNonAlliedProperties(myCo, gameMap);
+
+    // If we are already capturing any of these properties, remove them from the list.
+    for( Unit unit : myCo.units )
+    {
+      if( unit.getCaptureProgress() > 0 )
+      {
+        nonAlliedProperties.remove(unit.getCaptureTargetCoords());
+      }
+    }
+
+    // If the CO has enough AP, preload the CommanderAbilityAction.
+    ArrayList<CommanderAbility> abilities = myCo.getReadyAbilities();
```

Comment:
I feel like a basic abstraction that could help the AI a lot would be a flag or set of flags that says when the ability is useful.
Witchfire or Strong/Patch's abilities are definitely desirable at the start of the turn, but Sear is something you'd want to use only at the end of your turn.

I don't know what other flags we might want, but my heart sinks a little every time an AI pops Sear at the start of its turn, lol.